apps/managedidentity/managedidentity_test.go

```diff
@@ -48,95 +79,130 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
 }
 
 func makeResponseWithErrorData(err string, desc string) ([]byte, error) {
-	responseBody := ErrorRespone{
+	responseBody := ErrorResponse{
 		Err:  err,
 		Desc: desc,
 	}
 	jsonResponse, e := json.Marshal(responseBody)
 	return jsonResponse, e
 }
 
-type resourceTestData struct {
-	source   Source
-	endpoint string
-	resource string
-	miType   ID
+func createMockFile(t *testing.T, path string, size int64) {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		t.Fatalf("failed to create directory: %v", err)
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		t.Fatalf("failed to create file: %v", err)
+	}
+
+	if size > 0 {
+		if err := f.Truncate(size); err != nil {
+			t.Fatalf("failed to truncate file: %v", err)
+		}
+	}
+	f.Close()
 }
 
-type errorTestData struct {
-	code          int
-	err           string
-	desc          string
-	correlationid string
+func getMockFilePath(t *testing.T) (string, error) {
+	tempDir := t.TempDir()
+	mockFilePath := filepath.Join(tempDir, "AzureConnectedMachineAgent")
+
+	return mockFilePath, nil
 }
 
-func Test_SystemAssigned_Returns_AcquireToken_Failure(t *testing.T) {
-	testCases := []errorTestData{
-		{code: http.StatusNotFound,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusNotImplemented,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusServiceUnavailable,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusBadRequest,
-			err:           "invalid_request",
-			desc:          "Identity not found",
-			correlationid: "121212",
-		},
+func setEnvVars(t *testing.T, source Source) {
+	switch source {
+	case AzureArc:
+		t.Setenv(IdentityEndpointEnvVar, "http://127.0.0.1:40342/metadata/identity/oauth2/token")
+		t.Setenv(ArcIMDSEnvVar, "http://127.0.0.1:40342 value")
+	case AppService:
+		t.Setenv(IdentityEndpointEnvVar, "identityEndpointEnvVar value")
+		t.Setenv(IdentityHeaderEnvVar, "identityHeaderEnvVar value")
+	case CloudShell:
+		t.Setenv(MsiEndpointEnvVar, "msiEndpointEnvVar value")
+	case ServiceFabric:
+		t.Setenv(IdentityEndpointEnvVar, "identityEndpointEnvVar value")
+		t.Setenv(IdentityHeaderEnvVar, "identityHeaderEnvVar value")
+		t.Setenv(IdentityServerThumbprintEnvVar, "identityServerThumbprintEnvVar value")
+	}
+}
+
+func unsetEnvVars() {
```

Comment:
`t.Setenv("VARIABLE", "")` is better because it has no side effect on other tests